proptest/src/sugar.rs

```diff
@@ -164,6 +164,23 @@ macro_rules! proptest {
             }
         )*
     };
+
+    (#![proptest_config($config:expr)]
+     $(
+        $(#[$meta:meta])*
+       async fn $test_name:ident($($parm:pat in $strategy:expr),+ $(,)?) $body:block
+    )*) => {
+        $(
+            $(#[$meta])*
+            async fn $test_name() {
+                let mut config = $config.clone();
+                config.test_name = Some(
+                    concat!(module_path!(), "::", stringify!($test_name)));
+                $crate::proptest_helper!(@_BODY config ($($parm in $strategy),+) [] $body);
+            }
+        )*
+    };
```

Comment:
I think it should be possible to do this without the duplication by matching an optional `async` keyword via `$( async )?`, then outputting it in the same way.

Reply:
nevermind that's not a thing